Assistant: Add default model indicator in model picker

cameronmpalmer opened the pull request:
Fixes #11166

## Summary

Adds a visual indicator showing which model is configured as the default for each provider in the Assistant model picker dropdown.

When a user has configured `positron.assistant.models.preference.byProvider`, the default model for each provider now displays with a "(Default)" text suffix next to the model name.

**Before:**
<img width="365" height="380" alt="Screenshot 2025-12-22 at 11 50 44 AM" src="https://github.com/user-attachments/assets/206d3e55-f7d5-49c0-8fe0-8e043a30a2de" />


**After:**
(When "Haiku" is configured as default)

<img width="365" height="316" alt="Screenshot 2025-12-22 at 11 52 55 AM" src="https://github.com/user-attachments/assets/37b41f88-9ae5-4bce-b4db-997717c517a2" />



## Changes

- Modified `modelPickerActionItem.ts` to check `model.metadata.isDefault` property
- Added "(Default)" text suffix to model label when `isDefault` is true
- Enhanced tooltip to also show "(Default)" for consistency

## Implementation Details

The implementation leverages the existing `isDefault` property that's already being set by the backend (`extensions/positron-assistant/src/modelResolutionHelpers.ts`) based on user preferences. No backend changes were needed.

The frontend simply detects this property and adds the visual indicator:
```typescript
const label = isDefault ? `${model.metadata.name} (Default)` : model.metadata.name;
```

## Testing

Manually tested with:
- Setting `positron.assistant.models.preference.byProvider` with various models
- Switching between different defaults
- Multiple providers with different defaults
- Verified tooltip shows "(Default)" suffix

Screenshots attached showing the indicator in action.

### Release Notes

#### New Features

- Assistant model picker now shows a "(Default)" indicator next to the model configured as default for each provider

Commit:
Assistant: Add default model indicator in model picker

Add a "(Default)" text suffix to the model name in the model picker
dropdown to visually indicate which model is configured as the default
for each provider based on the positron.assistant.models.preference.byProvider
setting.

The indicator appears next to the model name (e.g., "Claude Haiku 4.5 (Default)")
and is also reflected in the tooltip. This helps users quickly identify their
configured default model when switching between models.

Fixes #11166

sharon-wang reviewed (commented):
Hey @cameronmpalmer! Would you be up for making an additional enhancement to move the default model to the top of the list? Something like this:

<img width="300" alt="Image" src="https://github.com/user-attachments/assets/37381994-4886-47c5-a4bf-839294c8abf3" />

This makes it a bit easier for a user to locate their default model for the provider, as some providers have long model lists and getting to the default model can take some scrolling.

Commit:
Assistant: Move default model to top of provider group in model picker

Sort each provider's models to place the default model first in the
dropdown list. This improves UX by making the default model immediately
visible without scrolling, especially for providers with long model lists.

cameronmpalmer commented:
Also, opened another PR https://github.com/posit-dev/positron/pull/11295 to merge a fix to a problem I found with the Echo provider while testing this change.
